syft/cataloger/cataloger.go

```diff
@@ -1,16 +1,48 @@
+/*
+Package cataloger provides the ability to process files from a container image or file system and discover packages
+(e.g. gems, wheels, jars, rpms, debs, etc.). Specifically, this package contains both a catalog function to utilize all
+catalogers in child packages as well as the interface definition to implement a specific cataloger.
+*/
 package cataloger
 
 import (
 	"github.com/anchore/stereoscope/pkg/file"
+	"github.com/anchore/syft/syft/cataloger/apkdb"
+	"github.com/anchore/syft/syft/cataloger/bundler"
+	"github.com/anchore/syft/syft/cataloger/dpkg"
+	"github.com/anchore/syft/syft/cataloger/golang"
+	"github.com/anchore/syft/syft/cataloger/java"
+	"github.com/anchore/syft/syft/cataloger/javascript"
+	"github.com/anchore/syft/syft/cataloger/python"
+	"github.com/anchore/syft/syft/cataloger/rpmdb"
 	"github.com/anchore/syft/syft/pkg"
 	"github.com/anchore/syft/syft/scope"
 )
 
+// Cataloger describes behavior for an object to participate in parsing container image or file system
+// contents for the purpose of discovering Packages. Each concrete implementation should focus on discovering Packages
+// for a specific Package Type or ecosystem.
 type Cataloger interface {
+	// Name returns a string that uniquely describes a cataloger
 	Name() string
-	// TODO: add ID / Name for catalog for uniquely identifying this cataloger type
+	// SelectFiles discovers and returns specific files that the cataloger would like to inspect the contents of.
 	SelectFiles(scope.FileResolver) []file.Reference
-	// NOTE: one of the errors which is returned is "IterationNeeded", which indicates to the driver to
-	// continue with another Select/Catalog pass
+	// Catalog is given the file contents and should return any discovered Packages after analyzing the contents.
 	Catalog(map[file.Reference]string) ([]pkg.Package, error)
+	// TODO: add "IterationNeeded" error to indicate to the driver to continue with another Select/Catalog pass
+	// TODO: we should consider refactoring to return a set of io.Readers instead of the full contents themselves (allow for optional buffering).
+}
+
+// All returns a slice of all locally defined catalogers (defined in child packages)
```

Comment:
Seems like you are using a dot at the end of every godoc string as a convention. This is a nit, but it would be nice to fully normalize to one or the other

Reply:
righto, attempting to keep them as sentences (unless it is a struct field comment). I'll adjust.